godotengine/godot
# Android: Remove permissions that have been removed from Android API

Addresses part 1 of https://github.com/godotengine/godot-proposals/discussions/9492

## Removal of permissions not existing in Android API

Looking at the diffs between Android API levels, we can see that some permissions have been removed. Since not present in the API anymore, they likely don't have a use case for most apps, and are tech debt, so we can remove them too. If there's a rare use case anyway, they can be added in the Custom Permissions array.

### API [23](https://developer.android.com/sdk/api_diff/23/changes/android.Manifest.permission): 

![Screenshot 2024-04-11 215841](https://github.com/godotengine/godot/assets/13484849/b93a9962-1a25-46d4-b724-071227566f01)

### API [24](https://developer.android.com/sdk/api_diff/24/changes/android.Manifest.permission)

![image](https://github.com/godotengine/godot/assets/13484849/c0a9926f-3dc3-4bad-aaf9-75ac3416431d)

### API [29](https://developer.android.com/sdk/api_diff/29/changes/android.Manifest.permission)

![image](https://github.com/godotengine/godot/assets/13484849/a4771ac4-04be-4dac-89d8-316f3065f647)

## HOW TO TEST
- build the engine
- go to Export for Android
- see that the removed permissions are gone from the list
